base/util.jl

```diff
@@ -450,36 +450,30 @@ if Sys.iswindows()
 function getpass(prompt::AbstractString)
     print(prompt)
     flush(stdout)
-    p = Vector{UInt8}(undef, 128) # mimic Unix getpass in ignoring more than 128-char passwords
-                          # (also avoids any potential memory copies arising from push!)
-    try
-        plen = 0
-        while true
-            c = ccall(:_getch, UInt8, ())
-            if c == 0xff || c == UInt8('\n') || c == UInt8('\r')
-                break # EOF or return
-            elseif c == 0x00 || c == 0xe0
-                ccall(:_getch, UInt8, ()) # ignore function/arrow keys
-            elseif c == UInt8('\b') && plen > 0
-                plen -= 1 # delete last character on backspace
-            elseif !iscntrl(Char(c)) && plen < 128
-                p[plen += 1] = c
-            end
+    s = SecretBuffer()
+    plen = 0
+    while true
+        c = ccall(:_getch, UInt8, ())
```

Comment:
the return type of `:_getch` is `Cint` (always a UInt8 value – it's documented to never return the expected EOF error code possibility)